rclpy/rclpy/qos.py

```diff
@@ -12,8 +12,108 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from enum import IntEnum
+from rclpy import _rclpy_implementation_rmw_agnostic
+
 
 class QoSProfile:
-    pass
+    """Class defining Quality of Service policies."""
+
+    __slots__ = [
+        '_history',
+        '_depth',
+        '_reliability',
+        '_durability',
+    ]
+
+    def __init__(self, **kwargs):
+        assert all(['_' + key in self.__slots__ for key in kwargs.keys()]), \
+            "Invalid arguments passed to constructor: %r" % kwargs.keys()
+        self.history = kwargs.get(
+            'history',
+            QoSHistoryPolicy.RMW_QOS_POLICY_HISTORY_SYSTEM_DEFAULT)
+        self.depth = kwargs.get('depth', int())
+        self.reliability = kwargs.get(
+            'reliability',
+            QoSReliabilityPolicy.RMW_QOS_POLICY_RELIABILITY_SYSTEM_DEFAULT)
+        self.durability = kwargs.get(
+            'durability',
+            QoSDurabilityPolicy.RMW_QOS_POLICY_DURABILITY_SYSTEM_DEFAULT)
+
+    @property
+    def history(self):
+        """Get QoSHistoryPolicy 'history'."""
```

Comment:
the suggestion was more to make it clearer that `QoSHistoryProfile` in this comment is referring to the type (when I got to [this line](https://github.com/ros2/rclpy/pull/39/files/2ca4e21a729efaafce76ae43dd7ced7d04bd7293#diff-8110981f890a0a14e8a04fb4d65d5832R75) it wasn't immediately obvious)